appcues/src/main/java/com/appcues/debugger/DebuggerViewModel.kt

```diff
@@ -247,31 +241,42 @@ internal class DebuggerViewModel(
         return toMutableList().onEach { it.showOnFab = false }
     }
 
-    fun onDetailDismiss() {
-        // reset the view state to remove any deep link path
-        _uiState.value = Expanded(Debugger(null))
-    }
+    fun captureScreen(): Capture? =
+        AppcuesActivityMonitor.activity?.captureScreen(config, contextResources)
 }
 
 private fun Activity.captureScreen(
     config: AppcuesConfig,
     contextResources: ContextResources,
-) {
-    val timestamp = Date()
-    val view = window.decorView.rootView
-    val displayName = view.screenCaptureDisplayName(timestamp)
-    val layout = view.asCaptureView()
-    layout?.let {
-        val capture = Capture(
-            appId = config.applicationId,
-            displayName = displayName,
-            screenshotImageUrl = null,
-            layout = layout,
-            metadata = contextResources.generateCaptureMetadata(),
-            timestamp = timestamp,
-        )
+): Capture? =
```

Comment:
this function could be on a separate file like "ScreenCaptor"  and it could be a dependency of DebuggerViewModel, what do you think?

Reply:
Yeah I like the idea of handling screen capture responsibilities in it's own class - I'm going to take a refactor pass at that. Thanks for the suggestions!